test_conformance/basic/test_get_linear_ids.cpp

```diff
@@ -59,7 +59,11 @@ test_get_linear_ids(cl_device_id device, cl_context context, cl_command_queue qu
 
 
     // Create the kernel
-    error = create_single_kernel_helper_with_build_options(context, &program, &kernel, 1, linear_ids_source, "test_linear_ids", "-cl-std=CL2.0");
+    std::string cl_std = "-cl-std=CL";
+    cl_std += (get_device_cl_version(device) == Version(3, 0)) ? "3.0" : "2.0";
```

Comment:
This could be made less fragile by using `>= Version(3, 0)`, i.e. for whenever some future version of OpenCL arrives.

If this sort of `-cl-std=CL2.0` test going to need fixed in more places or if this is it?   If there will be a lot maybe some kind of helper function that returns the highest supported OpenCL C version would be useful.  E.g. CL2.0 for 2.x devices, and the highest version in `CL_​DEVICE_​OPENCL_​C_​ALL_​VERSIONS` for >= 3.0 devices.

Reply:
In fact, the code in https://github.com/KhronosGroup/OpenCL-CTS/pull/879 could be the basis of that helper function?

Reply:
@alycm Thanks for the feedback. Yes, this assumption that `-cl-std=CL2.0` will need to be updated in quite a few places so I think having a helper function is a good idea. One thing I considered doing is adding this logic into the kernel helper framework where by it would automatically add `-cl-std` using the latest CL C version supported by the device to the list of compiler options if that option hadn't already been defined in the string, that way the calling code wouldn't have to specify the version each time.